org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/RunGradleJvmTestLaunchRequestJob.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2015 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Simon Scholz (vogella GmbH) - initial API and implementation and initial documentation
+ */
+
+package org.eclipse.buildship.core.launch;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.text.DateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.buildship.core.CorePlugin;
+import org.eclipse.buildship.core.console.ProcessDescription;
+import org.eclipse.buildship.core.i18n.CoreMessages;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.gradleware.tooling.toolingclient.Request;
+import com.gradleware.tooling.toolingclient.TestConfig;
+
+/**
+ * Executes tests through Gradle based on a given list of {@code IJavaElement}
+ * instances and a given set of {@code GradleRunConfigurationAttributes}.
+ */
+public final class RunGradleJvmTestLaunchRequestJob extends BaseLaunchRequestJob {
+
+    private final ImmutableList<IJavaElement> javaElements;
```

Comment:
The Job class should not know about JDT (just like the `RunGradleTestLaunchRequestJob` doesn't know about `OperationItem`s). 
The conversion should be part of the launch shortcut implementation (or in a helper class possibly).